src/AsyncTCP.cpp

```diff
@@ -724,22 +716,39 @@ bool AsyncClient::connect(IPAddress ip, uint16_t port){
     tcp_recv(pcb, &_tcp_recv);
     tcp_sent(pcb, &_tcp_sent);
     tcp_poll(pcb, &_tcp_poll, 1);
-    //_tcp_connect(pcb, &addr, port,(tcp_connected_fn)&_s_connected);
     _tcp_connect(pcb, _closed_slot, &addr, port,(tcp_connected_fn)&_tcp_connected);
     return true;
 }
 
+bool AsyncClient::connect(IPAddress ip, uint16_t port){
+    ip_addr_t addr;
+    addr.type = IPADDR_TYPE_V4;
+    addr.u_addr.ip4.addr = ip;
+
+    return _connect(addr, port);
+}
+
+bool AsyncClient::connect(IPv6Address ip, uint16_t port){
+    ip_addr_t addr;
+    addr.type = IPADDR_TYPE_V6;
+    memcpy(addr.u_addr.ip6.addr, static_cast<const uint32_t*>(ip), sizeof(uint32_t) * 4);
+
+    return _connect(addr, port);
+}
+
 bool AsyncClient::connect(const char* host, uint16_t port){
     ip_addr_t addr;
-    
+
     if(!_start_async_task()){
       log_e("failed to start task");
       return false;
     }
-    
+
     err_t err = dns_gethostbyname(host, &addr, (dns_found_callback)&_tcp_dns_found, this);
     if(err == ERR_OK) {
-#if LWIP_IPV4 && LWIP_IPV6
```

Comment:
I think the removal of this ifdef might be causing the build issues.

Reply:
That's what I saw too. I switched back to the previous version in my project.